ci/cloudbuild/builds/check-api.sh

```diff
@@ -81,57 +92,65 @@ function dump_abi() {
     zcat "${expected_dump_path}" >"cmake-out/${expected_dump_file}"
-    report="cmake-out/compat_reports/${lib}/src_compat_report.html"
+    report="cmake-out/compat_reports/${library}/src_compat_report.html"
     compliance_flags=(
       # Put the output report in a separate directory for each library
       -report-path "${report}"
       # We only want a source-level report. We make no ABI guarantees, such
       # as data structure sizes or virtual table ordering
       -src
-      # We ignore all symbols in internal namespaces, because these are not part
-      # of our public API. We do this by specifying a regex that matches against
-      # the mangled symbol names. For example, 8 is the number of characters in
-      # the string "internal", and it should again be followed by some other
-      # number indicating the length of the symbol within the "internal"
-      # namespace. See: https://en.wikipedia.org/wiki/Name_mangling
+      # We ignore all symbols in internal namespaces, because these are not
+      # part of our public API. We do this by specifying a regex that matches
+      # against the mangled symbol names. For example, 8 is the number of
+      # characters in the string "internal", and it should again be followed
+      # by some other number indicating the length of the symbol within the
+      # "internal" namespace. See: https://en.wikipedia.org/wiki/Name_mangling
       -skip-internal-symbols "(8internal|_internal)\d"
       # The library to compare
-      -l "${lib}"
+      -l "${library}"
       # Compared the saved baseline vs. the dump for the current version
       -old "cmake-out/${expected_dump_file}"
       -new "cmake-out/${actual_dump_file}"
     )
-    if ! io::run abi-compliance-checker "${compliance_flags[@]}"; then
-      io::log_red "ABI Compliance error: ${lib}"
-      ((++errors))
-      io::log "Report file: ${report}"
-      w3m -dump "${report}"
-    fi
+    abi-compliance-checker "${compliance_flags[@]}" >/dev/null || true
   fi
+
   # Replaces the (old) expected dump file with the (new) actual one.
   gzip -n "cmake-out/${actual_dump_file}"
   mv -f "cmake-out/${actual_dump_file}.gz" "${expected_dump_path}"
+}
+export -f check_abi # enables this function to be called from a subshell
+
+mapfile -t libraries < <(printf "google_cloud_cpp_%s\n" "${feature_list[@]}")
+
+# Run the check_abi function for each library in parallel since its slow.
+echo "${libraries[@]}" | xargs -P "$(nproc)" -n 1 \
+  bash -c "TIMEFORMAT=\"\${0#google_cloud_cpp_} completed in %0lR\";
+           time check_abi \${0} ${INSTALL_PREFIX} ${PROJECT_ROOT}"
+
+# A count of the number of libraries that fail the api compliance check.
+# This will become the script's exit code.
+errors=0
+
+for library in "${libraries[@]}"; do
+  report="cmake-out/compat_reports/${library}/src_compat_report.html"
+  if grep --silent "<td class='compatible'>100%</td>" "${report}"; then
+    io::log_green "ABI Compliance OK: ${library}"
```

Comment:
Ah, that is clever.  I could not figure out how to capture the status code from a call to `xargs`.